### PR TITLE
Allow installing packages with SSH in all node-workflows

### PR DESCRIPTION
**Pull request by shvlv**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature.

**What is the current behavior?** (You can also link to an open issue here)
We can't install packages with a `git+ssh` URL.

**What is the new behavior (if this is a feature change)?**
The approach is adopted from assets compilation workflows to configure the SSH agent. This way we can install packages from GitHub repositories directly.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

**Other information**:

### CODE REVIEW

File: .github/workflows/wp-scripts-lint.yml [hunk 1]
```diff
@@ -67,6 +76,9 @@ jobs:
     env:
       NODE_OPTIONS: ${{ inputs.NODE_OPTIONS }}
       NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+      GITHUB_USER_SSH_KEY: ${{ secrets.GITHUB_USER_SSH_KEY }}
+      GITHUB_USER_NAME: ${{ secrets.GITHUB_USER_NAME }}
+      GITHUB_USER_EMAIL: ${{ secrets.GITHUB_USER_EMAIL }}
```
Comment (tyrann0us) [inline .github/workflows/wp-scripts-lint.yml:81]: Why do we set these environment variables here if we read the values from the _secrets_ later and only use the env vars to check for existence (which is unnecessary in one of two cases)?
```suggestion
```
If you agree with this change, please also update accordingly in https://github.com/inpsyde/reusable-workflows/blob/main/.github/workflows/build-and-push-assets.yml. Thanks!

File: .github/workflows/wp-scripts-lint.yml [hunk 2]
```diff
@@ -82,6 +94,18 @@ jobs:
               .parse(process.env.ENV_VARS)
               .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
 
+      - name: Set up SSH
+        if: ${{ env.GITHUB_USER_SSH_KEY != '' }}
+        uses: webfactory/ssh-agent@v0.8.0
+        with:
+          ssh-private-key: ${{ secrets.GITHUB_USER_SSH_KEY }}
+
+      - name: Setup Git
```
Comment (tyrann0us) [inline .github/workflows/wp-scripts-lint.yml:103]:
```suggestion
      - name: Set up Git
```
Could you please fix this here as well, according to the pathfinder rules?
https://github.com/inpsyde/reusable-workflows/blob/f0fbb4ebdb4628bb40f9a949c3646720875e0e6a/.github/workflows/build-and-push-assets.yml#L116
Thanks!

File: .github/workflows/wp-scripts-lint.yml [hunk 3]
```diff
@@ -82,6 +94,18 @@ jobs:
...
+        if: ${{ env.GITHUB_USER_EMAIL != '' && env.GITHUB_USER_NAME != '' }}
```
Comment (tyrann0us) [inline .github/workflows/wp-scripts-lint.yml:104]: `GITHUB_USER_EMAIL` and `GITHUB_USER_NAME` are required, so there is no need to check them here again.
```suggestion
```

File: .github/workflows/wp-scripts-lint.yml [hunk 4]
```diff
@@ -82,6 +94,18 @@ jobs:
               .parse(process.env.ENV_VARS)
               .forEach(envVar => core.exportVariable(envVar.name, envVar.value));
 
+      - name: Set up SSH
+        if: ${{ env.GITHUB_USER_SSH_KEY != '' }}
```
[the remark this answers is not in the document]
Comment (shvlv) [reply .github/workflows/wp-scripts-lint.yml:98]: It does not work https://docs.github.com/en/actions/security-guides/encrypted-secrets#:~:text=Secrets%20cannot%20be%20directly%20referenced%20in%20if%3A%20conditionals.%20Instead%2C%20consider%20setting%20secrets%20as%20job%2Dlevel%20environment%20variables%2C%20then%20referencing%20the%20environment%20variables%20to%20conditionally%20run%20steps%20in%20the%20job.

File: .github/workflows/wp-scripts-lint.yml (hunk 3 above)
Comment (shvlv) [reply .github/workflows/wp-scripts-lint.yml:104]: yes, I copied most of the code from https://github.com/inpsyde/reusable-workflows/blob/f0fbb4ebdb4628bb40f9a949c3646720875e0e6a/.github/workflows/build-assets-compilation.yml. But secret definitions from https://github.com/inpsyde/reusable-workflows/blob/f0fbb4ebdb4628bb40f9a949c3646720875e0e6a/.github/workflows/build-and-push-assets.yml#L55-L63 :smile: . I will update PR to make all new secrets optional.

**Review by tyrann0us — COMMENTED** [review]
Thanks for the feedback. I know this changes the original scope of the PR, but as a pathfinder rule and to align all workflows, could you please 
1. add env variables for `GITHUB_USER_EMAIL` and `GITHUB_USER_NAME` in `build-and-push-assets.yml`
2. read `GITHUB_USER_*` from the respective env variables in `build-and-push-assets.yml` and `build-assets-compilation.yml` in the "Set up SSH" and "Set up Git" steps
3. sort the declaration of the `GITHUB_USER_*` env variables to follow the declaration of the secrets in all workflows 🙈  (i.e., `GITHUB_USER_EMAIL`, `GITHUB_USER_NAME`, `GITHUB_USER_SSH_KEY`)
4. Add `GITHUB_USER_*` secrets to _all_ remaining workflows (and documentations)

Thanks a ton!

Edit: Added point 4 after a discussion with @Chrico.

**Review by tyrann0us — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**